spec/models/mark_spec.rb

```diff
@@ -70,6 +73,166 @@
     end
   end
+
+  describe '#calculate_deduction' do
+    let(:assignment) { create(:assignment_with_deductive_annotations) }
+    let(:annotation_category_with_criteria) do
+      assignment.annotation_categories.where.not(flexible_criterion_id: nil).first
+    end
+    let(:result) { assignment.groupings.first.current_result }
+    let(:annotation_text) { annotation_category_with_criteria.annotation_texts.first }
+    let(:mark) { result.marks.first }
+
+    it 'calculates the correct deduction when one annotation is applied' do
+      deducted = mark.calculate_deduction
+      expect(deducted).to eq(1.0)
+    end
+
+    it 'calculates the correct deduction when multiple annotations are applied' do
+      create(:text_annotation,
+             annotation_text: annotation_text,
+             result: result)
+      deducted = mark.calculate_deduction
+      expect(deducted).to eq(2.0)
+    end
+
+    it 'returns 0 when override is true' do
+      mark.update!(mark: 1.0, override: true)
+      deducted = mark.calculate_deduction
+      expect(deducted).to eq(0)
+    end
+
+    it 'returns 0 when markable type is non flexible' do
+      rubric_criterion = create(:rubric_criterion, assignment: assignment)
+      non_flex_mark = create(:rubric_mark, markable: rubric_criterion)
+      deducted = non_flex_mark.calculate_deduction
+      expect(deducted).to eq(0)
+    end
+  end
+
+  describe '#update_deduction' do
+    let(:assignment) { create(:assignment_with_deductive_annotations) }
+    let(:annotation_category_with_criteria) do
+      assignment.annotation_categories.where.not(flexible_criterion_id: nil).first
+    end
+    let(:result) { assignment.groupings.first.current_result }
+    let(:annotation_text) { annotation_category_with_criteria.annotation_texts.first }
+    let(:mark) { result.marks.first }
+
+    it 'changes the mark correctly to reflect deductions when there are deductions with the same values' do
+      create(:text_annotation,
+             annotation_text: annotation_text,
+             result: result)
+      result.reload
+      expect(mark.mark).to eq(1.0)
+    end
+
+    it 'changes the mark correctly to reflect deductions when there are deductions with different values' do
+      new_text = create(:annotation_text_with_deduction,
+                        deduction: 1.5,
+                        annotation_category: annotation_category_with_criteria)
+      create(:text_annotation,
+             annotation_text: new_text,
+             result: result)
+      result.reload
+      expect(mark.mark).to eq(0.5)
+    end
+
+    it 'does not change the mark if override is enabled' do
+      mark.update!(mark: 3.0, override: true)
+      create(:text_annotation,
+             annotation_text: annotation_text,
+             result: result)
+      result.reload
+      expect(mark.mark).to eq(3.0)
+    end
+
+    it 'does not allow deductions to reduce the mark past 0' do
+      3.times do
+        create(:text_annotation,
```

Comment:
For duplicated object creation, use [`create_list`](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#building-or-creating-multiple-records).